target/linux/octeon/base-files/lib/preinit/01_sysinfo

```diff
@@ -1,31 +1,69 @@
+do_get_shield_mode() {
+        # Figure out what mode the Shield is in
+        # We need to determine what Mode the Shield is in - Time to query the GPIOs
+        GPIO=99
+        GPIO16=99
+        GPIO17=99
+
+        # GPIO16 decides Recovery (formally Gateway) Mode.
+        GPIO=16
+        echo ${GPIO} > /sys/class/gpio/export
+        echo in > /sys/class/gpio/gpio$GPIO/direction
+        GPIO16=$(cat /sys/class/gpio/gpio${GPIO}/value)
+        echo $GPIO > /sys/class/gpio/unexport
+
+        # GPIO17 decides between Bridge and Router Mode
+        GPIO=17
+        echo $GPIO > /sys/class/gpio/export
+        echo in > /sys/class/gpio/gpio$GPIO/direction
+        GPIO17=$(cat /sys/class/gpio/gpio${GPIO}/value)
+        echo $GPIO > /sys/class/gpio/unexport
+
+        if [ $GPIO16 -eq 0 ]; then
+           # Recovery Mode
+           echo "itusgateway"
+        else
+           if [ $GPIO17 -eq 0 ]; then
+              # Bridge Mode
+                 echo "itusbridge"
+           else
+              # Router Mode
+                 echo "itusrouter"
+           fi
+        fi
+}
```

Comment:
Is there really no better way?

Actually, if there is no better, I'd actually prefer to make three images, and have the user choose the correct one instead of this detection mechanism.

Reply:
This is in prep to moving these to their own Subtarget, which will not assign BOARD_NAME correctly, or that I couldn't.  This is how the device reads the multi-switch port to know what mode it's in.

Reply:
It could be inferred from the `root=` command line that u-boot provides, couldn't it? 

Or we could make u-boot put it somewhere in the device tree?

Reply:
the Initramfs files don't use root=, which is how the images get loaded onto the device.  

This code is what sets /tmp/sysinfo/board_name, so if I can do it here and not have to do with elsewhere in the system..  Or, can someone suggest a better way?

Reply:
So, please, explain what's actually going on here? Is this one device with three different modes or is it three devices?

Reply:
One device BOARD, but the switch determines which kernel image is chosen at boot, so THREE separate images (separate files, different root= defines, etc)

Reply:
So, at a reseller where you would buy it, there is actually only one "device" then?

Reply:
And the three images then just trigger different "configuration modes" for that some physical "device"?